pex/dist_metadata.py

```diff
@@ -508,7 +508,9 @@ def __str__(self):
         return self._str
 
 
-@attr.s(frozen=True)
+# N.B.: DistributionMetadata can have an expensive hash when a distribution has many requirements;
+# so we cache the hash. See: https://github.com/pantsbuild/pex/issues/1928
+@attr.s(frozen=True, cache_hash=True)
```

Comment:
This is the fix. All the rest just allows this to work. The attrs internals do magic when you enable this option and that magic requires being able to `import attrs`.